### ISSUE TITLE
configurable `/-/ready` healtcheck

### ISSUE THREAD

**Issue by WojciechKuk** [VictoriaMetrics/VictoriaMetrics#5250]
### Is your feature request related to a problem? Please describe

sometimes I need to restart one of my victoriametrics servers
during restart, metrics are buffered on vmagents (on other servers), and will arrive few minutes after successful startup

for some time after startup, victoriametrics serving data with a "hole" - until it accepts all data from vmagent queues

I need my loadbalancers to serve data only from full-ready backends, that have all datapoints in the database.
Not from restarted backends which will be updated with metrics in a few minutes.

### Describe the solution you'd like

I know, that it will be hard to estimate, that all datapoints has already been loaded and there is no lag, 

so I think it would be enough if victoriametrics can start publishing the "ready" status delayed by a configurable time after startup i.e.
`--delayedReadyTime=5m`

**Comment by Amper** [comment]
Hey @WojciechKuk.
Adding a `minReadySeconds` field to the operator and charts for statefulset-based VM entities will work for you?
I have a task like this in my works: https://github.com/VictoriaMetrics/helm-charts/issues/608

**Comment by hagen1778** [comment]
Hey @WojciechKuk!
If you use [replication](https://docs.victoriametrics.com/#replication) then there should not be data gaps. 
You might be interested in https://docs.victoriametrics.com/Cluster-VictoriaMetrics.html#improving-re-routing-performance-during-restart

**Comment by hagen1778** [comment]
Thanks for the explanation!

>when I restart one of the servers, it is delayed for 10-30 minutes compared to the others (queue on vmagent) - I don't want the loadbalancer to treat it as "healthy" during this time.

I don't see how vm-single could determine whether vmagents drained their queues already.
So it is more like setting a static value `--delayedReadyTime=5m` as proposed here https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5250#issue-1968692614. 

My position on this is the following: I don't see how `--delayedReadyTime=5m` is related to VM components. If such a setting could be handled by a 3rd party proxy/service which has 0 knowledge of VictoriaMetrics component specifics with the same level of success, then this setting should be handled by 3rd-party proxy/service.

Btw, I recalled that [vmauth](https://docs.victoriametrics.com/vmauth.html) has the following flag:
>   -failTimeout duration
     Sets a delay period for load balancing to skip a malfunctioning backend (default 3s)

Would it make sense to use vmauth with configured `-failTimeout=30m` to solve your case?

**Comment by WojciechKuk** [comment]
>If such a setting could be handled by a 3rd party proxy/service which has 0 knowledge of VictoriaMetrics component specifics with the same level of success
I agree, unfortunately sometimes there is no possibility to have influence on loadbalancer's configs.

but sooner or later there will be other use cases related to the fact that victoria always/immediately publishes healtcheck ready=OK (that's what it's for, to distinguish between transitional states)

(I will consult use of `vmauth` with with my colleagues)
